Tests/TuistLoaderTests/Loaders/RecursiveManifestLoaderTests.swift

```diff
@@ -258,6 +258,36 @@ final class RecursiveManifestLoaderTests: TuistUnitTestCase {
         ])
     }
 
+    func test_loadWorkspace_withSameProjectName() throws {
+        // Given
+        let workspace = Workspace.test(
+            name: "ProjectA",
+            projects: [
+                ".",
+            ]
+        )
```

Comment:
If I followed along, the bug we wasn't so much to do the the name of the workspace / project being similar, rather tuist 3.0.1 was acceidentally ignoring the Workspace manifest all together if it resided in the same directory!! 😅

As such the following modification to the workspace to make it distinct would help catch this.

```suggestion
        let workspace = Workspace.test(
            name: "MyWorkspace",
            projects: [
                ".",
            ],
            schemes: [
                Scheme(name: "CustomWorkspaceScheme")
            ]
        )
```

Reply:
That's my understanding as well, and the origin of [my question](https://github.com/tuist/tuist/pull/4236#discussion_r825309442) 🚀

Reply:
Right! The change in the Unit Test will make sure it's the actual "custom" workspace, thank you folks! I will merge once it pass @danyf90 @kwridan